src/drawing/polygon.rs

```diff
@@ -58,38 +53,41 @@ where
         );
     }
 
-    let mut y_min = i32::MAX;
-    let mut y_max = i32::MIN;
-    for p in poly {
-        y_min = min(y_min, p.y);
-        y_max = max(y_max, p.y);
-    }
+    let (mut y_min, mut y_max) = poly
```

Comment:
Nit: I find this much harder to read at a glance than the code it replaces. Don't feel strongly though.